solr/core/src/java/org/apache/solr/handler/RequestHandlerBase.java

```diff
@@ -51,9 +52,10 @@
 import static org.apache.solr.core.RequestParams.USEPARAM;
 
 /**
- *
+ * Base class for all request handlers.
  */
-public abstract class RequestHandlerBase implements SolrRequestHandler, SolrInfoBean, NestedRequestHandler, ApiSupport {
+public abstract class RequestHandlerBase implements
```

Comment:
would we want to have a default implementation returning ALL?

Reply:
Forcing each new request handler to make a permission decision ensures there is a concious decision and not just some default.
What does it even mean for an endpoint to declare the ALL permission?

Reply:
> So does the Schema screen requires "read" to show the screen? That's an interesting screen... it can expose data in terms of field values in term statistics but otherwise it's mostly a "config-read" tool. I don't recall how this screen maps to handlers.

I just checked what end points are queried when clicking around in the UI. Turns out that the collection-read and core-read are always queried to fill the collection/core dropdowns. It should of course as a SPA remember the values and not re-query every time, but I did not dive in to try to fix that.

The Schema screen uses Luke, and Luke requires the READ permission. I think it fetches all field names etc from luke as well, so the screen would be pretty useless without access to Luke.

Reply:
@madrob How does the `ALL` permission work? If we made it the default, how would that be different from a handler not implementing the `PermissionNameProvider` interface at all? And how is that again different from `null`?